app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.app.feature.crossselling.ui
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.activity.compose.setContent
+import com.hedvig.app.BaseActivity
+import kotlinx.parcelize.Parcelize
+import java.time.LocalDate
+
+class CrossSellingResultActivity : BaseActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val crossSellingResult: CrossSellingResult =
+            intent.getParcelableExtra(CROSS_SELLING_RESULT) ?: CrossSellingResult.Error
+        setContent {
+            CrossSellingResultScreen(crossSellingResult)
+        }
+    }
+
+    companion object {
+        fun newInstance(context: Context, crossSellingResult: CrossSellingResult): Intent {
+            return Intent(context, CrossSellingResultActivity::class.java).apply {
+                putExtra(CROSS_SELLING_RESULT, crossSellingResult)
+            }
+        }
```

Comment:
I'm a big fan of the functional syntax for functions in kotlin that simply have one return statement:

```suggestion
        fun newInstance(context: Context, crossSellingResult: CrossSellingResult) = Intent(context, CrossSellingResultActivity::class.java).apply {
                putExtra(CROSS_SELLING_RESULT, crossSellingResult)
            }
```